app/src/main/java/org/mifos/mobile/ui/recent_transaction/RecentTransactionScreen.kt

```diff
@@ -0,0 +1,97 @@
+package org.mifos.mobile.ui.recent_transaction
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
+import org.mifos.mobile.R
+import org.mifos.mobile.core.ui.component.EmptyDataView
+import org.mifos.mobile.core.ui.component.MifosProgressIndicator
+import org.mifos.mobile.utils.RecentTransactionUiState
+
+/**
+ * @author pratyush
+ * @since 19/3/24
+ */
+
+@Composable
+fun RecentTransactionScreen(
+    recentTransactionUiState: RecentTransactionUiState,
+    recentTransactionViewModel: RecentTransactionViewModel = hiltViewModel()
+) {
+    val isRefreshing by recentTransactionViewModel.isRefreshing.collectAsStateWithLifecycle()
+    val swipeRefreshState = rememberSwipeRefreshState(isRefreshing = isRefreshing)
+
+    SwipeRefresh(
+        state = swipeRefreshState,
+        onRefresh = { recentTransactionViewModel.refreshTransactionList() }
+    ) {
+        Column(
+            modifier = Modifier.fillMaxSize()
+        ) {
+            when (recentTransactionUiState) {
+                RecentTransactionUiState.EmptyTransaction -> {
+                    EmptyDataView(
+                        modifier = Modifier.fillMaxSize().verticalScroll(rememberScrollState()),
+                        icon = R.drawable.ic_error_black_24dp,
+                        error = R.string.recent_transactions
+                    )
+                }
+
+                is RecentTransactionUiState.Error -> {
+                    EmptyDataView(
+                        modifier = Modifier.fillMaxSize().verticalScroll(rememberScrollState()),
+                        icon = R.drawable.ic_error_black_24dp,
+                        error = R.string.recent_transactions
+                    )
+                }
+
+                is RecentTransactionUiState.LoadMoreRecentTransactions -> {
```

Comment:
Load more implementation is totally wrong here, you just need to focus to update the data source and single LazyColumn will handle everytime. 

Example if you have loaded 50 item and in second load you got 50 more item then just update the main data source . it will automatically recompose the screen items and load only changes items.